### DISCUSSION TITLE
How to integrate concurrencpp with external signals

### DISCUSSION THREAD

**Discussion by cnnblike** [David-Haim/concurrencpp#157] [Q&A]
hi, I'm now working on a project powered by libevent and you know how codebase can go messy if someone overuses the callback & reactor pattern.
I'm now looking into concurrencpp, I'm just wondering, how shall we integrate this with external signals, like some fd is ready for read, some time is up, etc.

**Comment by David-Haim** [comment]
You need to write some glue-code that pushes tasks to concurrencpp executors when a signal is emitted
does that answer your question?
